ui/app/src/services/content.ts

```diff
@@ -942,40 +943,20 @@ export function getChildrenByPath(
   );
 }
 
-export function copy(site: string, item: CopyItem): Observable<{ success: boolean }>;
-export function copy(site: string, path: string): Observable<{ success: boolean }>;
-export function copy(site: string, itemOrPath: string | CopyItem): Observable<{ success: boolean }> {
-  let item = typeof itemOrPath === 'string' ? { item: [{ uri: itemOrPath }] } : { item: [itemOrPath] };
-  return post(`/studio/api/1/services/api/1/clipboard/copy-item.json?site=${site}`, item, CONTENT_TYPE_JSON).pipe(
-    pluck('response'),
-    catchError(errorSelectorApi1)
-  );
-}
-
-export function cut(site: string, path: string): Observable<any> {
-  return post(
-    `/studio/api/1/services/api/1/clipboard/cut-item.json?site=${site}`,
-    { item: [{ uri: path }] },
-    CONTENT_TYPE_JSON
-  ).pipe(pluck('response'), catchError(errorSelectorApi1));
-}
-
-export function paste(site: string, path: string): Observable<string[]> {
-  return get(
-    `/studio/api/1/services/api/1/clipboard/paste-item.json?site=${site}&parentPath=${encodeURIComponent(path)}`
-  ).pipe(pluck('response', 'status'), catchError(errorSelectorApi1));
+export function paste(siteId: string, targetPath: string, clipboard: Clipboard): Observable<string[]> {
+  return postJSON('/studio/api/2/content/paste', {
+    siteId,
+    operation: clipboard.type,
+    targetPath,
+    item: getPasteItemFromPath(clipboard.sourcePath, clipboard.paths)
+  }).pipe(pluck('response', 'items'));
 }
 
-export function duplicate(site: string, path: string): Observable<string> {
-  let parentPath: any = path;
-  if (path.endsWith('index.xml')) {
-    parentPath = withoutIndex(path);
-  }
-  return copy(site, path).pipe(
-    switchMap(() => paste(site, getParentPath(parentPath))),
-    // Duplicate only does shallow copy, so return the copied path
-    pluck('0')
-  );
+export function duplicate(siteId: string, path: string): Observable<string> {
+  return postJSON('/studio/api/2/content/duplicate', {
+    siteId,
+    path
+  }).pipe(pluck('response', 'item'));
```

Comment:
Return the whole response object so we don’t have to change the API if later on there’s more to return